File: csharp/2013/the-spectacular-now-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "The Spectacular Now" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "The Spectacular Now",
                year: 2013,
                plot: "A hard-partying high school senior's philosophy on life changes when he meets the not-so-typical 'nice girl.'",
                rating: 7.1
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "The Spectacular Now",
                year: 2013
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}

File: python/2013/the-spectacular-now-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add The Spectacular Now to the database
movies.insert(
    title="The Spectacular Now", 
    year=2013, 
    plot="A hard-partying high school senior's philosophy on life changes when he meets the not-so-typical "nice girl."", 
    rating=7.1
)

# Confirm that the movie was added
movie = movies.select(
    title="The Spectacular Now", 
    year=2013
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    